### PR TITLE
ItemGroup: Update button focus styles to be more consistent

### PR DESCRIPTION
**Pull request by mikachan**
## What?

Small update to make the focus styles of `ItemGroup` component consistent with other button focus styles in the editor.

## Why?

Resolves this comment: https://github.com/WordPress/gutenberg/issues/50757#issuecomment-1589314311

## How?

Copies the button focus styles used in the block settings panel.

## Testing Instructions

Compare the focus states of the buttons in the Pattern Inserter and the Global Styles panel. The focus states should include a styled border.

## Screenshots or screencast 

| Before | After |
| ------- | ----- |
| image| image |

| Before | After |
| ------ | ----- |
| image | image |

### CODE REVIEW

File: packages/components/src/item-group/styles.ts
Comment (juanfra) [inline packages/components/src/item-group/styles.ts:28]: I believe the `box-shadow` used [on the inserter](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/inserter-list-item/style.scss#L84) has `inset`.

**Review by juanfra — COMMENTED** [review]
Thank you @mikachan!

Looks good, I left a comment about the styles that we're using on the inserter and the difference with these changes.

Before merging, could you also please add an entry to the [CHANGELOG](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CHANGELOG.md), under the "Enhancements" sub-section?

File: packages/components/src/item-group/styles.ts
Comment (mikachan) [reply packages/components/src/item-group/styles.ts:28]: I see the inserter uses `inset`, but I was trying to align to the styles used in the block settings panel, and these seem to use [these styles](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/sidebar-button/style.scss#L11) which do not use `inset`. I think the biggest difference with using `inset` is that it removes the border radius:

| Without Inset | With Inset |
| ------------- | ---------- |
| image | image |

In the [original issue](https://github.com/WordPress/gutenberg/issues/50757), it seems that the version with a border radius is preferable, but perhaps the inserter should also be updated to match these styles too.

Comment (juanfra) [reply packages/components/src/item-group/styles.ts:28]: Thank you, Sarah! That makes sense; I was considering the inserter as the reference.

Comment (mikachan) [reply packages/components/src/item-group/styles.ts:28]: No worries, I also first looked at the inserter styles and then noticed they were slightly different from the screenshots in the original issue. We could follow up with another PR to make the inserter buttons use the same focus style as well.

**Comment by mikachan** [comment]
Thanks for reviewing, @juanfra! I've updated the CHANGELOG now too.

**Review by juanfra — APPROVED** [review]
Thanks @mikachan! LGTM

File: packages/components/src/item-group/styles.ts
Comment (ciampo) [inline packages/components/src/item-group/styles.ts:27]: Have you considered using [`:focus-visible`](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible) instead, so that the focus ring only shows when interacting with the keyboard?

**Review by ciampo — COMMENTED** [review]
👋  Thank you for working on this!

cc'ing @jasmussen in case he had any feedback on this style change

File: packages/components/src/item-group/styles.ts
Comment (mikachan) [reply packages/components/src/item-group/styles.ts:27]: Yes, actually I initially was going to use `:focus-visible`, but it looks like the other buttons in the editor mostly use `:focus`. This doesn't mean it's the best way, but I was mainly trying to keep the styling consistent between the other editor buttons. Maybe all the buttons could be updated to use `:focus-visible`?

Comment (ciampo) [reply packages/components/src/item-group/styles.ts:27]: That could be something to keep in mind, yes. Let's wait for Joen's feedback , in case 😄

Comment (jasmussen) [reply packages/components/src/item-group/styles.ts:27]: Hey, thanks for the ping! Just for clarity, here's before:

before 1

before 2

Here's after:

after 1

after 2

In the after state, the focus style is now the correct color and the correct width (1.5px) in both places. However to Marco's point, it also now adds sticks the focus style when mouse navigating, which is especially jarring in these drilldown situations. I'd lean towards using focus visible here, it's already being applied for the back button in the site view, despite that lingering tooltip.

Comment (mikachan) [reply packages/components/src/item-group/styles.ts:27]: Thanks both! I agree that `:focus-visible` makes more sense here, I've opened up a small follow-up PR: https://github.com/WordPress/gutenberg/pull/51787

**Pull request merged**
